[QA] 연락처에 등록된 이미지로 프로필 이미지 수정

stopstone opened the pull request:
## 작업 내용
- 연락처에서 받아온 photoUri를 읽어 FriendInitRequest에 이미지 메타데이터를 포함하고, presigned URL로 실제 파일까지 업로드하도록 ContactImageReader, ImageUploader 추가하였습니다.
- FriendContactUIModel.toFriendInitItemRequest와 DefaultFriendRepository.initFriends를 개선해 사진이 있는 친구는 자동으로 서버에 이미지가 등록되도록 처리하였습니다.
- 홈 ContactItem, 친구 상세 FriendProfileScreen, 연락처 선택 리스트에 ImageLoader를 적용해 imageUrl이 있을 경우 원형 썸네일을 표시하고, 없을 때만 기존 플레이스홀더 사용하였습니다.
- 
## 확인 방법
- 연락처에서 사진이 있는 친구를 선택해 주기 설정 완료 → 
서버 초기화 API 호출이 성공하고, 로그에서 presigned 업로드가 200으로 떨어지는지 확인

- 홈 화면, 친구 상세 화면에서 방금 등록한 친구의 실제 프로필 이미지가 표시되는지 확인
- 이미지가 없는 친구는 예전처럼 회색 기본 이미지가 나오는지 확인

stopstone committed:
feat: 연락처 프로필 이미지 업로드 및 표시 기능 추가

- 연락처의 프로필 이미지를 가져와 Presigned URL을 통해 업로드하는 로직을 구현했습니다.
- `ContactImageReader`를 추가하여 연락처 이미지 URI로부터 이미지 데이터를 읽어옵니다.
- `ImageUploader`를 추가하여 OkHttp를 사용해 이미지를 업로드합니다.
- 친구 목록 및 홈 화면에서 `ImageLoader`를 사용하여 프로필 이미지를 표시하도록 수정했습니다.

stopstone committed:
refactor: 친구 프로필 이미지 로딩 방식 수정

- `ImageLoader`를 사용하여 친구의 프로필 이미지를 동적으로 로드하도록 수정했습니다.
- 이미지 로딩 실패 시 표시할 플레이스홀더와 에러 이미지를 설정했습니다.
- 프로필 이미지를 원형으로 표시하기 위해 `clip(CircleShape)`를 적용했습니다.

stopstone committed:
refactor: ImageUploader에 OkHttpClient 주입

stopstone committed:
refactor: 친구 프로필 이미지 업로드 병렬 처리

- `initFriends` 함수 내에서 친구 프로필 이미지를 업로드할 때, `coroutineScope`와 `launch`를 사용하여 각 이미지를 병렬로 업로드하도록 수정했습니다.
- 이를 통해 여러 친구를 한 번에 추가할 때 이미지 업로드 시간을 단축하고 성능을 개선했습니다.

rhkrwngud445 commented on Near/app/src/main/java/com/alarmy/near/local/contact/ContactImageReader.kt:
```diff
@@ -0,0 +1,49 @@
+        private val contentResolver: ContentResolver,
+    ) {
+        fun read(uriString: String): ContactImageData? {
+            val uri = runCatching { Uri.parse(uriString) }.getOrNull() ?: return null
```
요거 제 환경에서는 아래처럼 바꿔라고 경고가 뜨는데, 지석님은 혹시 해당 현상 없으셨을까요?
```kotlin
uriString.toUri() 
```

rhkrwngud445 commented on Near/app/src/main/java/com/alarmy/near/local/contact/ContactImageReader.kt:
```diff
@@ -0,0 +1,49 @@
+    val fileName: String,
+    val contentType: String,
+    val fileSize: Int,
+    val data: ByteArray,
```
byteArray가 값이 다르면 다른 객체로 인식해서 equals 및 hashCode를 재정의하라는 경고가 뜨네요!

해결 방법으로 option+ enter로 해당 코드를 자동으로 추가할 수 있습니다! 코드가 복잡해진다면 필요한 곳에서 byteArray를 맵핑해주는 방법이 있을 것 같아요!

stopstone replied:
감사합니다! equals, hashCode를 재정의 하는 코드가 크게 복잡하지 않아 우선 재정의 하는 방법으로 개선해보았습니다!

rhkrwngud445 commented on Near/app/src/main/java/com/alarmy/near/network/uploader/ImageUploader.kt:
```diff
@@ -0,0 +1,37 @@
+                    .put(requestBody)
+                    .build()
+            okHttpClient.newCall(request).execute().use { response ->
+                if (!response.isSuccessful) {
```
사소한 것인데
response.isSuccessful.not()으로 가독성을 올릴 수 있다고 생각해요!

반영은 안하셔도 됩니다!

stopstone replied:
디테일한 코멘트 감사합니다!
궁금한 것이 .not() 함수를 사용하는 것이 가독성이 더 좋아보이는 편인가요?
저는 오히려 코드가 길어질 수 있다고 생각하여 ! 연산자를 활용하는 편인데 어떻게 생각하시나요?

rhkrwngud445 replied:
개인 취향인데, 저는 자연어처럼 코드가 읽히는 것을 선호해서 not을 꽤 쓰는 편인 것 같아요!
함수명이 길어지는 것, 복잡한 필드 선언시에 타입을 명시하는 것처럼 가독성 이점의 맥락으로 쓰고 있습니다!

rhkrwngud445 reviewed (commented):
번거로운 작업이었을 텐데 고생하셨습니다!
IDE 경고 문구 위주로 코멘트 남겼는데 확인 부탁드려요!

stopstone committed:
feat: ContactImageData에 equals, hashCode 구현

`ContactImageData` 데이터 클래스에 `equals`와 `hashCode` 메서드를 오버라이드하여 객체 비교가 올바르게 동작하도록 수정했습니다.

stopstone committed:
refactor: 다음 연락 주기 표시 스트링템플릿 형식으로 변경
